Make it open *every* link in the default browser

piegamesde opened the pull request:
Except for tabs that are pinned.

Obviously, we should add a menu with settings since not all users might want this.

piegamesde committed:
Make it open *every* link in the default browser

Except for tabs that are pinned.

kumar303 commented on app/default_opener.py:
```diff
@@ -26,11 +26,11 @@ def encodeMessage(messageContent):
 while True:
     receivedMessage = getMessage()
-    subprocess.check_call(['open', receivedMessage['link']]);
+    subprocess.check_call(['/bin/chromium', receivedMessage['link']]);
```
How does this open a link in the default browser? In other words: what if the default browser is Firefox?

kumar303 commented on extension/background.js:
```diff
@@ -1,18 +1,54 @@
-  console.log("Port was disconnected. Error:", chrome.runtime.lastError);
+    console.log("Port was disconnected. Error:", chrome.runtime.lastError);
+    /* If we lost connection, simply reconnect. */
+    port = chrome.runtime.connectNative("default_opener");
```
This looks like a good idea to recover but what would happen if there is an unrecoverable error? Does it try to reconnect in a loop forever?

kumar303 commented on extension/background.js:
```diff
@@ -1,18 +1,54 @@
   }
 });
+
+/** I have no idea what this does */
```
[browserAction.onClick](https://developer.chrome.com/extensions/browserAction#event-onClicked) gets triggered when you click the extension icon in the Chrome toolbar. I can see how this would be useful.

kumar303 reviewed (commented):
Hi, thanks for taking the time to open a pull request. I missed the notification for this somehow so I'm just now seeing it. These look like some nice features but I had a few questions.

piegamesde committed:
Fix #1 and upgrade to python3

Replaced the subprocess package to launch the browser by the webbrowser package since the other didn't work reliably.
